cpu/esp8266/periph/rtt.c

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright (C) 2020 Gunar Schorcht
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_esp8266
+ * @ingroup     drivers_periph_rtt
+ * @{
+ *
+ * @file
+ * @brief       Low-level RTT driver implementation for ESP8266
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ *
+ * @}
+ */
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+#include "cpu.h"
+#include "log.h"
+#include "periph/rtt.h"
+
+#include "esp/common_macros.h"
+#include "esp/dport_regs.h"
+#include "esp/rtc_regs.h"
+#include "sdk/sdk.h"
+
+#define FRC2_CLK_DIV_256    2   /* divider for the 80 MHz AHB clock */
+
+#define RTC_BSS_ATTR __attribute__((section(".rtc.bss")))
+
+/**
+ * FRC2 is a 32-bit countup timer, triggers interrupt when reaches alarm value.
+ */
+typedef struct {
+    uint32_t load;
+    uint32_t count;
+    union {
+        struct {
+            uint32_t intr_hold : 1;
+            uint32_t reserved1 : 1;
+            uint32_t clk_div   : 2;
+            uint32_t reserved2 : 2;
+            uint32_t reload    : 1;
+            uint32_t enable    : 1;
+            uint32_t intr_sta  : 1;
+            uint32_t reserved3 : 23;
+        };
+        uint32_t val;
+    } ctrl;
+    union {
+        struct {
+            uint32_t clear    : 1;
+            uint32_t reserved1: 31;
+        };
+        uint32_t val;
+    } intr;
+    uint32_t alarm;
+} frc2_struct_t;
+
+extern volatile frc2_struct_t frc2;
+
+typedef struct {
+    uint32_t alarm;         /**< alarm */
+    rtt_cb_t alarm_cb;      /**< alarm callback */
+    rtt_cb_t overflow_cb;   /**< overflow callback */
+    void *alarm_arg;        /**< argument for alarm callback */
+    void *overflow_arg;     /**< argument for overflow callback */
+} rtt_config_t;
+
+static rtt_config_t rtt_config = { };
```

Comment:
`static` elements are already initialized with 0.

Reply:
You are right, static variables are placed in .bss section and are initialized with 0 therefore. So there is no need to assign a initializer.

Reply:
Changed.